crypto/pkcs7/pk7_smime.c

```diff
@@ -280,7 +280,8 @@ int PKCS7_verify(PKCS7 *p7, STACK_OF(X509) *certs, X509_STORE *store,
                     ERR_raise(ERR_LIB_PKCS7, ERR_R_X509_LIB);
                     goto err;
                 }
-                X509_STORE_CTX_set_default(cert_ctx, "smime_sign");
+                if (!X509_STORE_CTX_set_default(cert_ctx, "smime_sign"))
+                    goto err;
```

Comment:
Why do we not add an error to the stack here?

Reply:
We do not need because the error stack entry is already added by the function being called: `X509_STORE_CTX_set_default()`
I generally prefer this style because the called function knows much better what went wrong and why.